### PR TITLE
Chore/fluentd upgrade config improv

### PR DESCRIPTION
**Pull request by fauzigo**
Jira Ticket: [PXP-xxxx](https://ctds-planx.atlassian.net/browse/PXP-xxxx)
- [ ] Remove this line if you've changed the title to (PXP-xxxx): <title>

### Improvements
- Support for the latest version of fluentd.
- New fluentd configuration would also push worker nodes `messages` and `secure` logs.
- Tagging the logs in a simpler way for better visibility and manipulation through cloudwatch.
- Moving fluentd to its own namespace to clear out the kube-system namespace

### CODE REVIEW

File: gen3/bin/kube-setup-fluentd.sh [hunk 1]
```diff
@@ -16,15 +16,30 @@ ctx="$(g3kubectl config current-context)"
 ctxNamespace="$(g3kubectl config view -ojson | jq -r ".contexts | map(select(.name==\"$ctx\")) | .[0] | .context.namespace")"
 # only do this if we are running in the default namespace
 if [[ "$ctxNamespace" == "default" || "$ctxNamespace" == "null" ]]; then
-  if (! g3kubectl --namespace=kube-system get daemonset fluentd > /dev/null 2>&1) || [[ "$1" == "--force" ]]; then
```
Comment (frickjack) [inline gen3/bin/kube-setup-fluentd.sh:19]: should we undeploy the fluentd daemonset in `kube-system` if it exists?

File: gen3/bin/kube-setup-fluentd.sh [hunk 2]
```diff
@@ -16,15 +16,30 @@ ctx="$(g3kubectl config current-context)"
...
-      gen3 update_config fluentd-gen3 "${GEN3_HOME}/kube/services/fluentd/gen3.conf"
+      if (! g3kubectl get namepace logging > /dev/null 2>&1); then
+        g3kubectl apply -f "${GEN3_HOME}/kube/services/fluentd/fluentd-namespace.yaml"
+      fi
+      fluentdVersion="$(g3k_manifest_lookup '.versions["fluentd"]' "$manifestPath" |awk -F: '{print $2}')"
+      export KUBECTL_NAMESPACE=logging
+
+      if [ ${fluentdVersion} == "v1.10.2-debian-cloudwatch-1.0" ];
+      then
+        fluentdConfigmap="${GEN3_HOME}/kube/services/fluentd/gen3-1.10.2.conf"
```
Comment (frickjack) [inline gen3/bin/kube-setup-fluentd.sh:29]: Add a comment here - something like:
```
# for legacy backward compatability
```
so it's obvious which is the old, and which is the new ...

File: gen3/bin/kube-setup-fluentd.sh (hunk 1 above)
Comment (fauzigo) [reply gen3/bin/kube-setup-fluentd.sh:19]: The instruction is already there, perhaps you missed it https://github.com/uc-cdis/cloud-automation/pull/1238/files#diff-e9e44415744c2b19024b113d2d176dbaR37

File: gen3/bin/kube-setup-fluentd.sh (hunk 2 above)
Comment (fauzigo) [reply gen3/bin/kube-setup-fluentd.sh:31]: Comments added

**Pull request merged**